src/components/Portal/Portal.js

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useRef, useState } from 'react'
+import { createPortal } from 'react-dom'
+import PropTypes from 'prop-types'
+
+/**
+ * Hooks handles the creation and tear down of the portal's root
+ * element.
+ *
+ * @private
+ *
+ * @param {String} id - The id of the target container
+ *
+ * @return {HTMLElement} - The DOM node to use as the Portal target
+ */
+function usePortal(id) {
+  const [isAttached, setIsAttached] = useState(false)
+  const el = useRef(
+    document.getElementById(id) || document.createElement('div')
+  )
+
+  useEffect(() => {
+    setIsAttached(!!el.current.parentElement)
+
+    if (!isAttached) {
+      el.current.id = id
+      document.body.appendChild(el.current)
```

Comment:
Do we always want it attached to the `body` or should we accept an optional `container` element since you can ultimately do: `ReactDOM.createPortal(child, container)` and fallback default to `document.body`?

Reply:
So I think for dynamic portals we always want to attach to the `body`, which is the specific use case that this component solves. There might be a use case where we the `root` element is already rendered in the `DOM` (as static `html`), however I think in the majority of our cases, we want the `portal` to be created dynamically.

Reply:
Ok, well I've not used portals and so let's just go with what you've got here as you've been using this for real use cases and I trust you're more familiar with realistic usage. Thanks for clarifications Brad.